server/image_remove.go

```diff
@@ -10,15 +10,21 @@ import (
 	pb "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
 )
 
-// RemoveImage removes the image.
+// RemoveImage removes the image. If the image does not exist, then this
+// function should not error at all.
+//
+// For further documentation, see:
+// https://github.com/kubernetes/cri-api/blob/261df499b74595bc2ce546130f1edfcc6f05d2c1/pkg/apis/runtime/v1alpha2/api.proto#L123-L124
+// https://github.com/kubernetes/kubernetes/blob/71a7be41e0518a7df549eb1fa7b51e7c647d985b/pkg/kubelet/dockershim/docker_image.go#L122-L156
```

Comment:
Note that this implements something rather different in general: it removes all tags of the image and then presumably the image, whereas this implementation, via `UntagImage`, will just remove a tag(/digest?) if the input is a tag, leaving the image itself around.

(Assuming the comments at the link are correct and the input can only be an image ID, there probably isn’t a difference… but there are way too many abstraction layers that are way too liberal about the input they accept, so it’s hard to be sure.)